src/http/adapt.ml

```diff
@@ -85,5 +85,5 @@ let forward_body_h2
     response
     (H2.Body.Writer.write_string body)
     (H2.Body.Writer.write_bigstring body)
-    (H2.Body.Writer.flush body)
+    (fun f -> H2.Body.Writer.flush body (fun _ -> f ()))
```

Comment:
Doesn't this change make it incompatible with h2<0.14.0 ?

Reply:
`flush` takes `unit -> unit` before 0.13 and `[`Something | `Something] -> unit` with 0.13, but `fun _ -> f ()` is of type `'a -> unit` so it typechecks with both version (tried locally).

Reply:
OK, you might also need to add a lower bound, see https://github.com/ocaml/opam-repository/pull/27202/files

Reply:
Could indeed reproduce that 0.8.0 doesn’t work. Pushed a new version of the commit.